client/state/preferences/actions.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_REMOVE,
+	PREFERENCES_FETCH
+} from 'state/action-types';
+
+export const USER_SETTING_KEY = 'calypso_preferences';
+
+export function fetchPreferences() {
+	return ( dispatch ) => {
+		dispatch( { type: PREFERENCES_FETCH } );
+		return wpcom.me().settings().get().then( ( data ) => {
+			dispatch( {
+				type: PREFERENCES_RECEIVE,
+				data
+			} );
+		} );
+	};
+}
+
+const save = () => ( dispatch, getState ) => {
+	const settings = {};
+	settings[ USER_SETTING_KEY ] = getState().preferences.values;
+	wpcom.me().settings().update( JSON.stringify( settings ) ).then( ( data ) => {
+		dispatch( {
+			type: PREFERENCES_RECEIVE,
+			data
```

Comment:
When updating settings, only the updated values are returned in the response, not the entire set. Based on the reducer implementation below, I might expect that this would incorrectly wipe out all other preferences?

Including unit tests would help identify these issues earlier.